### DISCUSSION TITLE
How to get photos in the "biography" section?

### DISCUSSION THREAD

**Discussion by theoctavist** [TT-ReBORN/Georgia-ReBORN#128] [Q&A] [answered]
Re Display>>>*Artist View* vs *Album View* while t is cycling through the last.fm and all music info, quite often the photo area is blank except for the text "NO PHOTO"  I assumed that whatever processes were pulling the actual bio data (from last fm etc) would also pull the photos.  (last.fm certainly has plenty of artist photos)  I also know  that this is user error. I cant figure it out though.   help? 
![2023-06-23 18_05_45-Promotion - foobar2000](https://github.com/TT-ReBORN/Georgia-ReBORN/assets/5534014/43de0840-283b-4e07-b83d-9877de732a53)

**Comment by TT-ReBORN** [comment]
Hi @theoctavist,

the biography should automatically download and display all images from last.fm and allmusic.
You are really the first one to report this kind of issue, otherwise many people would already mention it.
It's hard to say what's going on, it could be multiple factors:

What's your OS?
What Georgia-ReBORN version are you using? I guess v3.0-RC1 and not from master?
Did you install foobar as portable or standard installation?
Did you change the location of the biography cache folder?
Did you change any settings at all or are you using the default ones?
How do you play music when your playlist is empty? Did you remove it or changed playlist?

What do you mean by "quite often the photo area is blank" ? 
Do you get any photos on different artists at all or is everything displayed with the NO PHOTO placeholder?

I would try to do a new portable foobar install and see if it changes or try on another computer.
If you have a firewall, it is possible that it could block the image fetch process. 

-TT

**Comment by d3rb** [comment]
Hi Tom, i have the same Problem, i see only the "No Foto" Placeholder on last.fm.
Fresh install in Portable and v3.0-RC1
![image](https://github.com/TT-ReBORN/Georgia-ReBORN/assets/15198548/980ef03f-84b1-4a0d-b718-ce2bdcb6a09b)

**Reply by TT-ReBORN** [reply]
Hi,

this is really strange, just tested it myself with a fresh install v.3.0-RC1 and latest foobar and have no problems. 
As I've mentioned above, could it be your firewall blocking it? Do you have enough hdd space?
Does it happen for all artists you're playing?
Do you have any images in your foobar2000\profile\cache\biography\biography-cache\art_img ?
Do you have a created "Bruce Springsteen" folder in foobar2000\profile\cache\biography\biography-cache\art_img\b\ ?

Can you zip your portable foobar and put the first song of Bruce Springsteen - 01. Born In The U.S.A into the zip 
and send it to me?

**Reply by TT-ReBORN** [reply]
Got your files, thanks!

**Comment by TT-ReBORN** [comment, marked as answer]
### PLEASE CHECK IF YOUR FIREWALL HAS NOT BLOCKED SOMETHING RELATED TO YOUR FOOBAR!

---

OK guys,

it is possible that last.fm or allmusic "HAD" some kind of server issues when you tried to fetch and download the bio images.
If you encounter the "NO PHOTO" placeholder in the biography, it's due to no photos were downloaded in your biography cache:
`foobar2000\profile\cache\biography\biography-cache\art_img`
There are only created directories with the artist names, but they are empty with only the update.txt file inside. 
The update.txt file is a flag that prevents to fetch and download all images again every time you play the same artist. 
It is to go easy on the last.fm/allmusic servers and also save performance.

**Solution:**
- If you have a lot of blank artists with only the update.txt, delete your whole biography-cache directory `foobar2000\profile\cache\biography`. On next foobar reload the biography cache will be recreated and all images 
  should be downloaded and displayed.
- If you only have some artist with "NO PHOTO" placeholder, right click somewhere in the biography panel for 
  context menu > Sources > Force update.
  
---  
  
@Wil-B,

it would be the best for your next biography version, to implement a method to check when artist directories do not have any photos,
then it should automatically force update. What do you think?

-Tom

**Reply by d3rb** [reply]
Hi Tom, sorry for the late reply, I was on vacation. I set everything up again and the Force update method doesn't work either. The art_img/b/Bruce Springsteen folder only contains the update.txt, no matter which album or artist. when i manually add photos in folder art_img/b/Bruce Springsteen artist1,artist2 i get the photos.but the download of the photos does not work

**Reply by TT-ReBORN** [reply]
Hi,

this is really weird because I tried your version you have send me and it works just fine. 
I have also tried other artist without problems, all images are downloaded successfully...
This could be a problem on your PC but we need to confirm this, try to download the old Beta 12d 
( it has the old biography version )
Here is the [link](https://www.file.io/6GCe/download/jmhe3A69mdby). 

Now in your existing portable foobar where the biography download does not work, just rename the profile folder to "profile_nobio" and copy the profile from the Beta 12d zip in your foobar.
Start and try again and see if biography image download now works. If it also does not work, it's a problem on your PC.
Report to me how it goes.

-Tom

**Reply by d3rb** [reply]
Hello Tom I found the problem the cscript.exe was actually blocked. my mistake sorry now it works many thanks
greetings
Enrico

**Reply by TT-ReBORN** [reply]
I'm glad you solved this Enrico, this issue had me worried a bit ;-)

**Comment by theoctavist** [comment]
Hey there! It's fixed.  It was an internet connection issue. Mea culpa

**Reply by TT-ReBORN** [reply]
No problem, good it's working for you.
